algorithms/src/snark/marlin/mod.rs

```diff
@@ -26,14 +42,28 @@ pub use circuit_verifying_key::*;
 mod errors;
 pub use errors::*;
 
-/// A generic implementation of the Marlin proof system.
+/// RNGs for the Marlin SNARK.
+pub mod fiat_shamir;
+pub use fiat_shamir::*;
+
+/// Implements the base Marlin zkSNARK proof system.
 mod marlin;
 pub use marlin::*;
 
 /// Specifies the Marlin mode.
 mod mode;
 pub use mode::*;
 
+/// Parameters of non-native field gadget
+///
+/// Sample parameters for non-native field gadgets
+/// - `BaseField`:              the constraint field
+/// - `TargetField`:            the field being simulated
+/// - `num_limbs`:              how many limbs are used
+/// - `bits_per_limb`:          the size of the limbs
+///
+pub mod params;
```

Comment:
I guess this can move back to the nonnative module?

Reply:
Open to suggestions how to handle. This was the tricky thing I mentioned. The Optimization type and params are used in Marlin it seems, so this would require snarkvm-algorithms to import snarkvm-gadgets which will slow down compile times

Reply:
Maybe we move it to snarkvm_utilities or something?

Reply:
(Also feel free to push directly if you have an idea in mind)

Reply:
Maybe into snarkvm-r1cs?

Reply:
that would also work. I guess also fine to leave it here; at some point soon we'll move to lookup-based non-native anyway =)